pkg/handler/v2/application_installation/application_installation.go

```diff
@@ -0,0 +1,193 @@
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(getApplicationInstallationReq)
+
+		client, err := userClientFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		applicationInstallation := &appskubermaticv1.ApplicationInstallation{}
+		if err := client.Get(ctx, types.NamespacedName{Namespace: req.Namespace, Name: req.ApplicationInstallationName}, applicationInstallation); err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		return convertInternalToExternal(applicationInstallation), nil
+	}
+}
+
+func UpdateApplicationInstallation(userInfoGetter provider.UserInfoGetter) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(updateApplicationInstallationReq)
+
+		client, err := userClientFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		// first fetch the currentAppInstall to make sure it exists
+		currentAppInstall := &appskubermaticv1.ApplicationInstallation{}
+		if err := client.Get(ctx, types.NamespacedName{Namespace: req.Namespace, Name: req.ApplicationInstallationName}, currentAppInstall); err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		newAppInstall := convertExternalToInternal(&req.Body)
+		currentAppInstall.Spec = newAppInstall.Spec
+
+		err = client.Update(ctx, currentAppInstall)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		return convertInternalToExternal(currentAppInstall), nil
+	}
+}
+
+func convertInternalToExternal(app *appskubermaticv1.ApplicationInstallation) *apiv2.ApplicationInstallation {
+	return &apiv2.ApplicationInstallation{
+		ObjectMeta: apiv1.ObjectMeta{
+			CreationTimestamp: apiv1.Time(app.CreationTimestamp),
+			Name:              app.Name,
+		},
+		Namespace: app.Namespace,
+		Spec:      &app.Spec,
+	}
+}
+
+func convertExternalToInternal(app *apiv2.ApplicationInstallation) *appskubermaticv1.ApplicationInstallation {
+	return &appskubermaticv1.ApplicationInstallation{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       appskubermaticv1.ApplicationInstallationKindName,
+			APIVersion: appskubermaticv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      app.Name,
+			Namespace: app.Namespace,
+		},
+		Spec: *app.Spec,
+	}
+}
+
+func userClientFromContext(ctx context.Context, userInfoGetter provider.UserInfoGetter, projectID, clusterID string) (ctrlruntimeclient.Client, error) {
+	clusterProvider := ctx.Value(middleware.ClusterProviderContextKey).(provider.ClusterProvider)
+	userInfo, err := userInfoGetter(ctx, projectID)
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	cluster, err := clusterProvider.Get(ctx, userInfo, clusterID, &provider.ClusterGetOptions{})
+	if err != nil {
+		return nil, err
```

Comment:
Is there a reason why you don't map it to HTTP error?
```suggestion
		return nil, common.KubernetesErrorToHTTPError(err)
```

Reply:
oh good point! I missed that one!